src/execution/sql/functions/string_functions.cpp

```diff
@@ -9,19 +9,25 @@
 
 namespace terrier::execution::sql {
 
-void StringFunctions::Concat(StringVal *result, exec::ExecutionContext *ctx, const StringVal &left,
-                             const StringVal &right) {
-  if (left.is_null_ || right.is_null_) {
-    *result = StringVal::Null();
-    return;
+void StringFunctions::Concat(StringVal *result, exec::ExecutionContext *ctx, const int64_t num_inputs,
+                             const StringVal *inputs[]) {
+  TERRIER_ASSERT(num_inputs != 0, "Concat should have at least one argument");
+
+  std::size_t length = 0;
+  for (int64_t i = 0; i < num_inputs; i++) {
+    if (!inputs[i]->is_null_) {
+      length += inputs[i]->GetLength();
+    }
   }
 
-  const std::size_t length = left.GetLength() + right.GetLength();
   char *const ptr = ctx->GetStringAllocator()->PreAllocate(length);
+  for (int64_t i = 0, offset = 0; i < num_inputs; i++) {
```

Comment:
Technically, `offset` should be a `std::size_t`. I'd also go ahead and change the type on line 17 for cosistency.

Reply:
Fixed